### ISSUE TITLE
how to obtain get past_kv_cache values? 

### ISSUE THREAD

**Issue by arunpatro** [OpenNMT/CTranslate2#1392]
In hugging face autoclass models, we can do `model.forward` to obtain both logits and past_key_values. The `model.generate` then uses these past_key_values for efficient gen. 

How can I return these past_key_values in `generator.forward_batch` api?

**Comment by guillaumekln** [comment]
At this time `forward_batch` is not designed to be used for iterative decoding. It computes the logits (or log probs) for a full sequence.

Why can't you use `generate_batch` for generation?

**Comment by arunpatro** [comment]
I definitely can use `generate_batch`, that works fast! But if I had access to the kv cache or HF like api then I could do fast guided generation like with the outlines. See: [PR](https://github.com/normal-computing/outlines/pull/207
)

**Comment by japuiglaucia** [comment]
how to obtain get past_kv_cache values? #1392
